### PR TITLE
TfLite stablehlo add convolution op

### PR DESCRIPTION
**Pull request by LokeshReddyOVS-MCW**
- Adds initial version of stablehlo_convolution op
- Adds basic integer type unit test

### CODE REVIEW

File: tensorflow/lite/kernels/stablehlo_convolution_test.cc [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    return ExtractVector<T>(output_);
+  }
+
+ protected:
+  int lhs_;
+  int rhs_;
+  int output_;
+};
+
+TEST(StablehloConvolutionOpTest, kBF16TestTypesTensorsWork) {
```
Comment (rascani) [inline tensorflow/lite/kernels/stablehlo_convolution_test.cc:107]: Here and below, we should drop the k prefix and just call it ConvolutionFloat32, ConvolutionBFloat16, ConvolutionFloat16, etc. Test names should summarize the behavior being tested and their expected behavior. Its also fine to omit the happy path behavior from the name.

File: tensorflow/lite/kernels/test_util.h [hunk 2]
```diff
@@ -37,11 +39,10 @@ limitations under the License.
 #include <utility>
 #include <vector>
 
-#include <gmock/gmock.h>
-#include <gtest/gtest.h>
 #include "absl/algorithm/container.h"
 #include "absl/types/span.h"
 #include "flatbuffers/flatbuffers.h"  // from @flatbuffers
+#include "tensorflow/core/platform/bfloat16.h"
```
Comment (rascani) [inline tensorflow/lite/kernels/test_util.h:45]: This isn't needed anymore.

File: tensorflow/lite/kernels/stablehlo_convolution_test.cc (hunk 1 above)
Comment (nishantsarda-mcw) [reply tensorflow/lite/kernels/stablehlo_convolution_test.cc]: changed as suggested

File: tensorflow/lite/kernels/test_util.h (hunk 2 above)
Comment (nishantsarda-mcw) [reply tensorflow/lite/kernels/test_util.h]: removed.

File: tensorflow/lite/kernels/BUILD [hunk 3]
```diff
@@ -1426,8 +1425,7 @@ cc_test(
 
 cc_test(
     name = "stablehlo_convolution_test",
-    srcs = ["stablehlo_convolution_test.cc"],
-    tags = ["tflite_nnapi"],
+srcs = ["stablehlo_convolution_test.cc"],
```
Comment (rascani) [inline tensorflow/lite/kernels/BUILD:1428]: Needs indentation

File: tensorflow/lite/kernels/test_util.h [hunk 4]
```diff
@@ -15,6 +15,8 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
 #define TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
 
+#include <gmock/gmock.h>
```
Comment (rascani) [inline tensorflow/lite/kernels/test_util.h:18]: Please move these lines back to where they were.

File: tensorflow/lite/kernels/test_util.h [hunk 5]
```diff
@@ -1353,4 +1358,4 @@ inline DimsAreMatcher DimsAre(absl::Span<int> dims) {
 
 }  // namespace tflite
 
-#endif  // TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
+#endif  // TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
```
Comment (rascani) [inline tensorflow/lite/kernels/test_util.h:1361]: Revert this change.

**Review by rascani — COMMENTED** [review]
Mostly LGTM! Just a couple minor nits left.

File: tensorflow/lite/kernels/test_util.h (hunk 4 above)
Comment (nishantsarda-mcw) [reply tensorflow/lite/kernels/test_util.h]: ok changed to original format.
Don't know why clang format make these changes.
